src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import duke.command.Command;
+
+/**
+ * The main program
+ */
+public class Duke {
+    private final Storage storage;
+    private final TaskList taskList;
+    private final Ui ui;
+
+    /**
+     * The main program's constructor.
+     *
+     * @param filePath The path to the data file.
+     */
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
```

Comment:
"this." keyword not needed, since there is no conflict in name.

Reply:
A fair point, but I like consistency, so all my instance variables use `this`.